railties/lib/rails/generators/rails/app/templates/Gemfile.tt

```diff
@@ -56,6 +56,10 @@ group :development do
 <%- end -%>
   # Speed up commands on slow machines / big apps [https://github.com/rails/spring]
   # gem "spring"
+
+<%- if RUBY_VERSION >= "3.1" -%>
```

Comment:
One minor thing: this template is mostly executed when creating a new app. So people upgrading either Rails or Ruby likely won't have this.

It would be nice if the error page would tell you you can add `error_highlight, 0.4` in your Gemfile if you are on Ruby >= 3.1 and the right error_highlight isn't loaded yet.

Reply:
Sounds good. I have implemented it. What do you think?

<img width="354" alt="image" src="https://user-images.githubusercontent.com/21557/186110046-81704653-3475-4759-ac13-a4ffc37f3022.png">

Do you want to include this change in this PR? I'm not too sure about the CSS or wording. Should I create another PR?

Here is a patch:

```diff
diff --git a/actionpack/lib/action_dispatch/middleware/templates/rescues/_source.html.erb b/actionpack/lib/action_dispatch/middleware/templates/rescues/_source.html.erb
index 9f168357ed..0ab2967fe4 100644
--- a/actionpack/lib/action_dispatch/middleware/templates/rescues/_source.html.erb
+++ b/actionpack/lib/action_dispatch/middleware/templates/rescues/_source.html.erb
@@ -28,6 +28,9 @@
           </tr>
         </table>
       </div>
+      <%- if defined?(ErrorHighlight) && Gem::Version.new(ErrorHighlight::VERSION) < Gem::Version.new("0.4.0") -%>
+        <p class="error_highlight_tip">Tip: You may want to add <code>gem 'error_highlight', '&gt;= 0.4.0'</code> into your Gemfile, which will display the fine-grained error location.</p>
+      <%- end -%>
     </div>
   <% end %>
 <% end %>
diff --git a/actionpack/lib/action_dispatch/middleware/templates/rescues/layout.erb b/actionpack/lib/action_dispatch/middleware/templates/rescues/layout.erb
index 4230b43742..0a8face483 100644
--- a/actionpack/lib/action_dispatch/middleware/templates/rescues/layout.erb
+++ b/actionpack/lib/action_dispatch/middleware/templates/rescues/layout.erb
@@ -154,6 +154,12 @@
       text-decoration: #F00 wavy underline;
     }

+    .error_highlight_tip {
+      color: #666;
+      padding: 2px 2px;
+      font-size: 10px;
+    }
+
     .button_to {
       display: inline-block;
       margin-top: 0.75em;
```

Reply:
Looks great to me. No need for a second PR.

Reply:
The change is merged into this PR. Thank you!